core/columns_api.php

```diff
@@ -73,6 +73,7 @@
 function columns_filter_disabled( array $p_columns ) {
 	$t_columns = array();
 	$t_enable_profiles = ( config_get( 'enable_profiles' ) == ON );
+	$t_enable_due_date = access_has_project_level( config_get( 'due_date_view_threshold' ) );
```

Comment:
Should we only make the call to `access_has_project_level()` when $p_columns include the corresponding fields?

Reply:
I'm with @vboctor here, better move this to the switch case, I believe it's better to risk running the access check twice (considering that due date and overdue are not so frequently used, and likely not simultaneously as they are redundant). 

And I guess the same reasoning applies to os/platform as well, although in this case it may be worth using a static var to cache the config value as the fields are often used together.